### PR TITLE
Users can create new auctions

### PR DESCRIPTION
**Pull request by seoulection**
* Also adds guard in dashboard against non-logged-in users

### CODE REVIEW

File: ebai-front/src/components/CreateAuctionForm.vue [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      title: '',
+      description: '',
+      startingBidPrice: 1,
+      buyItNowPrice: null,
+      endDate: null,
+      error: ''
+    }
+  },
+  computed: {
+    tomorrowsDate: function () {
```
Comment (omarnyte) [inline ebai-front/src/components/CreateAuctionForm.vue]: Nice use of computed properties!

File: ebai-front/tests/integration/SigninForm.spec.js [hunk 2]
```diff
@@ -6,6 +6,25 @@ import { mockRouter, resolvePromise } from '../helpers/helpers.js'
 jest.mock('@/api/users.js')
 
 describe('submitting the signup form', () => {
+  let wrapper
```
Comment (omarnyte) [inline ebai-front/tests/integration/SigninForm.spec.js]: I think the describe description should read "submitting the signin form".

File: ebai-front/tests/unit/auctions.spec.js [hunk 3]
```diff
@@ -28,4 +28,29 @@ describe('auctions.js', () => {
...
+    const apiResponse = await createAuction({data: "hey"})
+
+    expect(apiResponse.status).toEqual(201)
+  })
+
+  it('throws an error when auction is not created', async () => {
+    axios.post.mockRejectedValueOnce(new Error('This is an error'))
+
+    await expect(createAuction({data: "hello"})).rejects.toThrow('Auction cannot be saved. Please try again!') 
+  })
```
Comment (omarnyte) [inline ebai-front/tests/unit/auctions.spec.js]: I have the same thoughts about this sort of unit test as I do for the API unit tests in the previous PR. I really liked your integration test for CreateAuctionForm. I think that if you tweaked that integration test slightly so that it returned the stubbed data *only* for a `POST` to  `http://localhost:3000/auctions`, you might not even need this unit test. You'd probably need to use a library (like [fetch-mock](https://github.com/wheresrhys/fetch-mock)) to accomplish that since Jest can only mock so much. If you decide to do that, let me know if you need any help!

File: ebai-front/src/api/auctions.js [hunk 4]
```diff
@@ -1,5 +1,19 @@
...
+  }
 }
+
+export async function createAuction(data) {
+  try {
+    return await axios.post('http://localhost:3000/auctions', data, {
+      withCredentials: true
+    })
+  } catch(e) {
+    throw new Error('Auction cannot be saved. Please try again!')
```
Comment (kkrull) [inline ebai-front/src/api/auctions.js]: I like the part that says that the auction can't be saved, but it is often useful to include the message from the error you just caught too.  This is especially true for back end code.  Here in the front-end, there's more of a need to be judicious about what sort of technobabble you're showing to people on the web.

TL;DR - try to say something slightly more detailed that's still useful to the end user, if you can

File: ebai-front/tests/integration/SignupForm.spec.js [hunk 5]
```diff
@@ -16,74 +35,26 @@ describe('submitting the signup form', () => {
...
-      firstNameInput.setValue('Hello')
-      const lastNameInput = wrapper.find('#lastName')
-      lastNameInput.setValue('World')
-      const emailInput = wrapper.find('#email')
-      emailInput.setValue('hello@world.gov')
-      const passwordInput = wrapper.find('#password')
-      passwordInput.setValue('Hello1234%')
-      wrapper.find('form').trigger('submit.prevent')
-
+      fillOutForm(wrapper)
```
Comment (kkrull) [inline ebai-front/tests/integration/SignupForm.spec.js]: Nice use of adding the helper function, like we talked about in the earlier PRs

File: ebai-front/tests/unit/auctions.spec.js (hunk 3 above)
Comment (kkrull) [reply ebai-front/tests/unit/auctions.spec.js]: Agreed - sometimes it's easier to drive out and test certain kinds of behavior with an integration test.  Testing what happens with HTTP requests is a good example where this often comes up, and there are a number of possible options that make this an interesting question:

* mock out the function that makes the HTTP request and mock out its response: this tends to get into really low level details really quickly, increasing the chance of API Drift.
* still work with the notion of requests and responses, but do it more at a library level: this is how Angular tests tend to be written -- it has a special test function that lets you say what the response is
* fire up an instance of express.js from inside your javascript test and make a real request to it on a randomized port: tends to cause people to freak out, but useful sometimes
* avoid unit tests and focus more on integration tests, using the real UI and a real instance of your back end: handy approach for a small number of tests, but best saved for cases where other tests provide comprehensive coverage on everything else, and the only remaining unknown is whether the UI is configured with the right URL (wiring/configuration management)

TL;DR - if a test is really hard or feels really brittle, it might help to try another test strategy or introducing an abstraction
